src/commands/force/user/create.ts

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as os from 'os';
+import {
+  Aliases,
+  AuthInfo,
+  Connection,
+  DefaultUserFields,
+  fs,
+  Logger,
+  Messages,
+  Org,
+  SfdxError,
+  User,
+  UserFields,
+} from '@salesforce/core';
+import { get } from '@salesforce/ts-types';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-user', 'create');
+
+interface SuccessMsg {
+  name: string;
+  value: string;
+}
+
+interface FailureMsg {
+  name: string;
+  message: string;
+}
+
+export class UserCreateCommand extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('examples').split(os.EOL);
+  public static readonly requiresUsername = true;
+  public static readonly requiresDevhubUsername = true;
+  public static readonly varargs = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    alias: flags.string({
+      char: 'a',
+      description: messages.getMessage('flags.alias'),
+    }),
+    definitionfile: flags.string({
+      char: 'f',
+      description: messages.getMessage('flags.definitionfile'),
+    }),
+  };
+  public logger: Logger;
+  public org: Org;
+
+  private user: User;
+  private successes: SuccessMsg[];
+  private failures: FailureMsg[];
+
+  public async run(): Promise<UserFields> {
+    this.logger = await Logger.child(this.constructor.name);
+    const defaultUserFields: DefaultUserFields = await DefaultUserFields.create({
+      templateUser: this.org.getUsername(),
+    });
+    const user: User = await User.create({ org: this.org });
+
+    // merge defaults with provided values with cli > file > defaults
+    const fields: UserFields = await this.aggregateFields(defaultUserFields.getFields());
+
+    try {
+      await user.createUser(fields);
+    } catch (e) {
+      await this.catchCreateUser(e, fields);
+    }
+
+    // because we overload the UserField in the aggregateFields method these entries could possibly be there
+    const permsets: string = fields['permsets'];
```

Comment:
I don't like this. Can we change the return type of aggregateFields to `UserFIeld & Dictionary<string>`?